src/core/customresolver/customresolver.go

```diff
@@ -0,0 +1,75 @@
+package customresolver
```

Comment:
I don't like this change for multiple reasons:
- it's a bit over engineered. I can see what it does with caching and sharing that cache between goroutines but it has too much code just for that
- it's hardcoded just for fight with russian propaganda. I actually mean it when I say that this project can be used by actual red teams to load test their own infra. This change would make it harder
- going back to performance benefits of caching - you'll actually get max performance benefits by not doing any dns lookups at all and it already allows you to do that by resolving your dns outside db1000n (see statil_host in http client). This will allow you even more than dns caching as it will work across processes on different machines and also there are types of defense that can be bypassed by bypassing dns if improperly configured. This is what we do in the default config and there's a lot of recon going to that part that is not visible here
- Finally if you do want to have a way to dynamically query dns while running the app but share the results between multiple goroutines it's better to be achieved by having a "lookup_host" kind of template (I'm not sure if I don't have it already) and using more complex scenario in the config (storing dns result in a variable and using it in a sequence). You can even make a version of the template that would accept the list of dns servers to use. This makes the least amount of assumptions about average target and leaves most control capabilities to the one configuring the app

Reply:
> as it will work across processes on different machines

Well, custom DNS cilent makes it more universal - for different machines. 
Without this DNS client [from stoppropaganda], golang does not resolve DNS on Android.

Reply:
Hmm, the android issue is interesting but if you specify raw IPs in the config (outsourcing the DNS work out of the app) you won't encounter it

Reply:
that being said, I'd probably be ok with having custom dns that would work on android but it would be better to have it as a template function (Ideally having this dns as a separate module with minimum ties to the main package). Template approach has benefits of being more configurable and it would also not require doing any synchronization work as it would be possible to configure it to do only one dns resolution per target in the config